Rock Paper Scissor With Python ( Workshop Bounty )

JackTDC opened the pull request:
Criteria:

[x] Help club members produce a compelling and real project in under 20 minutes
Based on the test meeting, people were able to complete this workshop in under 20 minutes.

[x] Are very easy to hack on for long periods of time after completion
Some examples are listed at the end of the workshop as to how people can extend their projects and learn more things.

[x] Have instructive and compelling images, GIFs, and visuals
This workshop does contain instructive and compelling images, GIFs, and visuals.

[x] Use repl.it or a similar service that allows members to begin coding with one click.
Repl.it is used for this workshop.

[x] Link unknown concepts to external sources.
Linked a video to the typewriter effect.

[x] End with at least 3 demos of how the project can be expanded, with live links and code links for each demo.
Have provided some examples which people have created.

Good Workshop Guidelines:

[x] Workshop that meets all of the "Good workshops" guidelines outlined above.

[x] 1-2 minute video clip from your test club meeting showing what members built.
Here's the video clip from the demo meeting. 
Link: https://hackclub.slack.com/files/U016YD1ST61/F01BHL78ADD/gmt20201005-071240_arpit-pawa_2080x1170__online-video-cutter.com_.mp4

[x] All assets (images, PDFs, and any other binaries) must be hosted using the #cdn channel on the Hack Club Slack for posterity and to cut down on repository size.
All assets are hosted using the #cdn channel.

[x] All workshops must be in English at native-level proficiency. Phrasing and grammar must be perfect.

[x] You must be either a registered Hack Club leader or an active member of the Slack.
Club leader @ St. Mary's High School

faisalsayed10 commented:
Starting my review in a few seconds!

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+---
+Name: 'Rock Paper Scissors Game'
+Description: 'Easy rock paper scisoors game using Python'
```
```suggestion
Description: 'Simple rock paper scissors game using Python'
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+![Image](https://cloud-931ehqeec.vercel.app/coding.gif)
+
+### 1)Importing Modules
+Let us start making our game! First, we'll need to import some modules
```
```suggestion
Let us start making our game! First, we'll need to import some modules.
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+
+### 1)Importing Modules
+Let us start making our game! First, we'll need to import some modules
+So in the first line type `from random import randint`. This will import the radient module
```
```suggestion
So in the first line, type
```py
from random import randint
```
This will import the randint module to your project.
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+## Part 3: Making The Game
+![Image](https://cloud-931ehqeec.vercel.app/coding.gif)
+
+### 1)Importing Modules
```
```suggestion
### 1) Importing Modules
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+Let us start making our game! First, we'll need to import some modules
+So in the first line type `from random import randint`. This will import the radient module
+
+### 2)Giving values to the variables
```
```suggestion
### 2) Giving values to the variables
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+So in the first line type `from random import randint`. This will import the radient module
+
+### 2)Giving values to the variables
+First let's make a list for all th entries a plyer could enter,
```
```suggestion
First let's make a list for all the entries a player could enter,
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+First let's make a list for all th entries a plyer could enter,
+`t = ["r","p","s"]`
+
+Now we have to write a command so the computer chooses rock, paper or scissors randomly!
```
```suggestion
Now we have to write a code so the computer chooses rock, paper or scissors randomly!
```

faisalsayed10 replied:
Its a code actually and not a command

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+`t = ["r","p","s"]`
+
+Now we have to write a command so the computer chooses rock, paper or scissors randomly!
+`computer = t[randint(0,2)]`.
```
```suggestion
```py
computer = t[randint(0,2)]
```
```

JackTDC replied:
I didn't understand the change? can you explain

faisalsayed10 replied:
As it is a ONE LINE code, so i have suggested that u make it a code block. Also, the 'py' written there is syntax highlighting.

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+Now we have to write a command so the computer chooses rock, paper or scissors randomly!
+`computer = t[randint(0,2)]`.
+
+Now we will set the player to false so we can keep the game in a loop to do that just type, `Player = False`. Now you have to write some code so the player could enter his/her name. 
```
```suggestion
Now we will set a variable to false and by using it in a `while` loop, we can keep the game playing infinitely! To do that , just type `Player = False`. Now you have to write some code so the player can enter his/her name. 
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+`computer = t[randint(0,2)]`.
+
+Now we will set the player to false so we can keep the game in a loop to do that just type, `Player = False`. Now you have to write some code so the player could enter his/her name. 
+Also to make a score system enter the following code:
```
```suggestion
Also to make a score system, we'll write the following code:
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+
+player = False
+
+name = input("Enter your name:")
```
There was no description for this code? if it is a beginner, he/she/they might get confused as how it suddenly got ended up here. Please make sure you check that.

JackTDC replied:
line 68?

faisalsayed10 replied:
yes, u have not mentioned anything above related to getting input

sampoder replied:
^^^^^^

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+### 3)Keeping the game in a loop
+![Image](https://cloud-nmra250be.vercel.app/loop.gif)
+
+For this game I will using `while` loop but you can also use `for` loop! 
```
```suggestion
For this, I'll use a `while` loop but you can also use `for` loops! 
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+
+For this game I will using `while` loop but you can also use `for` loop! 
+
+So for starters type,
```
```suggestion
So we'll just type,
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
````diff
@@ -0,0 +1,274 @@
+
+Enter the following code:
+```
+while player == False:
````
This is becoming a huge chunk of code, you might split it into small parts and explain them separately.

JackTDC replied:
Most of the code is repetition so explaining it separately will just make the workshop wordy

faisalsayed10 replied:
If most of the 'SAME' code is repeatng, then i suggest u to make it a seperate function and call that function at required places

faisalsayed10 replied:
otherwise this code is too lengthy to be written in one go

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
````diff
@@ -0,0 +1,274 @@
+```
+#### Explanation:
+
+1. The first if statement declares a tie.
````
Also, just like above, please make it more elaborate.

JackTDC replied:
Most of the code is repetition so explaining it separately will just make the workshop wordy

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+4. Last else statement occurs only when the player enters a invalid word.
+
+### 5) Keeping the loop going
+Now as we have completed codin the game we need it to be looped infinite number of times.
```
```suggestion
Now as we have completed building the game we need it to run infinitely.
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
````diff
@@ -0,0 +1,274 @@
+
+To do this you just need to add the following statement after the `while` statement,
+```
+player = False
````
I think this is not required as u have already defined the player to be false previously, yeah as im doubtful about it, please verify that

JackTDC replied:
This keeps the loop going if you remove this the game will close itself after 1 round

faisalsayed10 replied:
Ok sure! I just had a doubt. No problem at all!

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
````diff
@@ -0,0 +1,274 @@
+
+And at last, you need to add the following command so the computer chooses a different outcome every time
+```
+computer = t[randint(0,2)]
````
this was defined previously too right?

JackTDC replied:
but this wasn't in al loop so if we remove this one the computer will choose only one thing every time (you can remove it and see)

faisalsayed10 replied:
No problem at all!

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+Here are some things which you can do:
+
+- Consider changing and adding more features!
+- Making the game a two player game.
```
```suggestion
- Making it a two player game.
```

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+
+- Consider changing and adding more features!
+- Making the game a two player game.
+- Make the game only of 3 turns.
```
```suggestion
- Make the game only of 3 turns instead of running it infinitely.
```

faisalsayed10 reviewed (changes requested):
This is a pretty fun workshop but this needs few improvements, please make the suggested changes and we'll see what next!

faisalsayed10 commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+2. Then, post the link to the [`#scrapbook`](https://hackclub.slack.com/messages/scrapbook) channel to share it with everyone!
+
+[slack]: https://slack.hackclub.com/
+
```
are these extra spaces below this line? u may remove them if they aren't required! :smile:

faisalsayed10 commented:
@sampoder can u please take a look too?

sampoder commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+
+# Rock Paper Scissor Game
+
+Have you ever thought about building a game with Python? Well today is your lucky day.
```
clearer intro

sampoder commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+
+You should see something like the following:
+
+![Image](https://cloud-lukrtynqs.vercel.app/repl.png)
```
actual alt text please

sampoder commented on workshops/rock_paper_scissors/README.md:
```diff
@@ -0,0 +1,274 @@
+
+### 1) Importing Modules
+Let us start making our game! First, we'll need to import some modules.
+So in the first line, type `from random import randint`. This will import the radient module.
```
spelling

sampoder commented:
this is a lot of fun! could you please add more detailed code explanations though?

JackTDC commented:
@sampoder I have made all the mentioned changes.

faisalsayed10 commented:
@sampoder any updates on this? I'm pretty much satisfied.

sampoder reviewed (changes requested):
How come you've made a clone of Fayd's workshop could you change that back

sampoder reviewed (changes requested):
Please also check the formatting, grammar and phrasing of the workshop

lachlanjc reviewed (changes requested):
+1 on Sam’s comments—I notice a bunch of grammatical errors, such as “make sure you do so to save this wonderful piece of creation”. It’d be great to link to external resources, such as the randint package. Also, this is missing syntax highlighting on the code blocks

JackTDC commented:
Ok @lachlanjc @sampoder I used grammarly to fix the grammatical mistakes, added links to external resources(randint and typewriter effect), and added syntax highlighting

sampoder reviewed (changes requested):
Check https://youtu.be/GZGIATuAne8, we're getting there but I got super lost in the middle of the workshop

JackTDC commented:
@sampoder I changed everything mentioned int the video some point I wanted to tell you

1)You said there is no draw system but there us a if statement where if the computer entry and player entry are same then it prints tie
2)Fayd changed the code by adding for loop and making the game last on three rounds, Vedant changes the code to last only 3 rounds but using while loop and CrazyBlackStone was a beginner in python so there aren't any massive changes   

And sorry for my poor writing

sampoder commented:
I don't fully understand 1 & 3.

But re 2:

These demos are meant to be considerable modifications of the original projects, view: https://workshops.hackclub.com/tunes/#going-beyond.

JackTDC commented:
the 3rd point I was talking about was about the if else statement
around 4:15 in the video U said to clean them up

and the 1st point is about draw situations,u mentioned in the video that there isn't any condition if the player entry and computer entry is the same (around 4:24 in the video) but there is a condition where it prints tie

sampoder commented:
Ahh gotcha, must have missed the tie part my bad

sampoder commented:
With the if else statements I would look into doing something with JSON, but that may be a bit complex so lets leave it I guess

JackTDC commented:
@sampoder I actually wanted to do that too but then Fayd told me that it should be beginner friendly so I ended up doing if else statement

faisalsayed10 commented:
Yeah, but if you are confident that you can explain that concept properly then go ahead with the JSON

JackTDC commented:
@faisalsayed10 I think If else statement is good here

faisalsayed10 commented:
:man_shrugging:

MatthewStanciu commented:
Hey, this is awesome! Thanks so much for submitting it! I think it's _almost_ there, just a few small changes that I notice that I think you should make and then I think it'll be ready to merge:

* The code doesn't quite work as intended. Line 5 of the program sets the computer input to a random value, but that value never changes, so if the computer selects "Paper", it will keep selecting paper for the entire game. (At least that was my experience playing with it on repl.it). I think you should update the code to have the computer choose a different action every time, and also update that part of the workshop accordingly
* I love that you included 3 specific examples of hacks, but quickly skimming through/playing with each of the hacks, I didn't notice anything very different. I'm sure there are differences, but these differences need to be pretty different from the original project and very easily noticed in order for them to make up a good "hacking" section.

I'm going to close this for now, but I think this is like 90% of the way there and I totally think you should make those few changes and resubmit. Ping me on Slack if you need any guidance :) excited to merge this soon!
